src/hashcomp.cpp

```diff
@@ -255,9 +255,19 @@ bool irc::tokenstream::GetToken(long &token)
 	return returnval;
 }
 
+unsigned int irc::tokenstream::Count()
+{
+	return this->numparts;
+}
+
 irc::sepstream::sepstream(const std::string& source, char separator, bool allowempty)
-	: tokens(source), sep(separator), pos(0), allow_empty(allowempty)
+	: tokens(source)
+	, sep(separator)
+	, pos(0)
+	, allow_empty(allowempty)
 {
+	this->tokenlen = this->tokens.length();
+	this->numparts = std::count(this->tokens.begin(), this->tokens.end(), this->sep);
```

Comment:
This is not valid because a separator may exist one or more times when allow_empty is false.

e.g. "foo  bar          baz" is a space delimited string with three tokens; "foo", "bar", and "baz".